GoogleDataTransport/GDTCORLibrary/GDTCOREvent.m

```diff
@@ -68,15 +58,26 @@ - (nullable instancetype)initWithMappingID:(NSString *)mappingID target:(NSInteg
   if (mappingID == nil || mappingID.length == 0 || target <= 0) {
     return nil;
   }
+  __block NSNumber *eventID;
+  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+  [GDTCOREvent nextEventIDForTarget:target
+                         onComplete:^(NSNumber *_Nullable newEventID) {
+                           eventID = newEventID;
+                           dispatch_semaphore_signal(sema);
+                         }];
+  if (dispatch_semaphore_wait(sema, dispatch_time(DISPATCH_TIME_NOW, 5 * NSEC_PER_SEC)) != 0) {
```

Comment:
I think we will need to find a way to avoid blocking the thread, but it doesn't look like a blocker to me.

Just to have a better context: why assigning `eventID` by storage (as it was implemented originally) doesn't work?

Reply:
The previous mechanism worked, but I'd rather the saving of data be consolidated to one place. That way, it can also be reset or deleted when necessary.

Also, the previous mechanism was blocking as well, so this isn't different than what is currently shipping. We have gotten some complaints about high cpu usage here, so it might be worth making it async in the future (I'm not sure I want to tackle that this PR, since there's no degradation in performance at least).